sparta/sparta/kernel/SpartaHandler.hpp

```diff
@@ -251,6 +316,45 @@ namespace sparta
     sparta::SpartaHandler::from_member<clname, &clname::meth>            \
     (this, #clname"::"#meth"()")
 
+    /*!
+     * \def CREATE_CONST_SPARTA_HANDLER(clname, meth)
+     *
+     * Creates a \c SpartaHandler type given the class name (\c
+     * clname) and a class method (\c meth).
+     *
+     * This method requires a class instance (expects \c this to be
+     * defined and of type \c clname).  The class method given must
+     * match the following signature:
+     *
+     * \code
+     * void clname::meth();
+     * \endcode
+     *
+     * Example:
+     *
+     * \code
+     * class MyClass
+     * {
+     * public:
+     *
+     *     MyClass() {
+     *         sparta::SpartaHandler handler =
+     *             CREATE_CONST_SPARTA_HANDLER(MyClass, myMethod);
+     *         // ... do something with the handle
+     *         handler();
+     *     }
+     *
+     *     void myMethod() {
+     *         std::cout << "myMethod called" << std::endl;
+     *     }
+     *
+     * };
+     * \endcode
+     */
+#define CREATE_CONST_SPARTA_HANDLER(clname, meth)                            \
```

Comment:
Did you really need a separate macro?

Reply:
No. Got confused there for a bit. Changing.

Reply:
My latest commit is not really what you wanted but it's a start.
I removed the need for Sparta Handlers from registerConsumerHandler method of Ports.
Tested this on core example.